manifests/components/kibana.jsonnet

```diff
@@ -38,18 +39,69 @@ local strip_trailing_slash(s) = (
     },
   },
 
+  // List of Kibana plug-ins to install
+  plugins:: {
+    /* Example:
+    logtrail: {
+      version: "v0.1.31",
+      url: "https://github.com/sivasamyk/logtrail/releases/download/v0.1.31/logtrail-6.7.1-0.1.31.zip",
+    },
+    */
+  },
+
   es: error "elasticsearch is required",
 
   serviceAccount: kube.ServiceAccount($.p + "kibana") + $.metadata {
   },
 
+  pvc: kube.PersistentVolumeClaim($.p + "kibana-plugins") + $.metadata {
+    storage: "2Gi",
+  },
+
   deploy: kube.Deployment($.p + "kibana") + $.metadata {
     spec+: {
       template+: {
         spec+: {
           securityContext: {
             fsGroup: 1001,
           },
+          volumes_+: {
+            plugins: kube.PersistentVolumeClaimVolume($.pvc),
+          },
+          initContainers_+: {
+            kibana_plugins_install: kube.Container("kibana-plugins-install") {
+              image: KIBANA_IMAGE,
+              securityContext: {
+                allowPrivilegeEscalation: false,
+              },
+              env_+: {
+                local plugins = [{name: name} + $.plugins[name] for name in std.objectFields($.plugins)],
+                KIBANA_PLUGINS: std.join('\n', ["%(name)s,%(version)s,%(url)s" % plugin for plugin in plugins]),
+              },
+              command: [
+                "/bin/bash",
+                "-c",
+                |||
+                  set -e
+                  set -x
+                  exec &> >(tee /opt/bitnami/kibana/plugins/log)
+                  rm -rf /opt/bitnami/kibana/plugins/lost+found
+                  for i in ${KIBANA_PLUGINS}
+                  do
+                    IFS=',' read -ra PLUGIN <<< "$i"
```

Comment:
This is fine, but if you want to avoid bash features for some reason (/bin/sh might be a lighterweight shell), then you could also do:
```
IPFS=,; set -- $PLUGIN; IPFS=
/opt/bitnami/kibana/bin/kibana-plugin remove --silent $1 || :
/opt/bitnami/kibana/bin/kibana-plugin install --no-optimize $3
```